--- conan/api/subapi/export.py
+++ conan/api/subapi/export.py
@@ -1,4 +1,6 @@
+import os
+import shutil
 from conan.api.output import ConanOutput
 from conan.internal.conan_app import ConanApp
 from conans.client.cmd.export import cmd_export
@@ -43,6 +45,15 @@
         conanfile.folders.set_base_folders(source_folder, output_folder)
         dest_package_folder = pkg_layout.package()
         conanfile.folders.set_base_package(dest_package_folder)
+        
+        local_pkg_metadata = conanfile.package_metadata_folder
+        dest_pkg_metadata = pkg_layout.metadata()
+        conanfile.folders.set_base_pkg_metadata(dest_pkg_metadata)
+        
+        if local_pkg_metadata and os.path.exists(local_pkg_metadata):
+            shutil.copytree(local_pkg_metadata, dest_pkg_metadata, dirs_exist_ok=True)
+        else:
+            from conans.util.files import mkdir
+            mkdir(dest_pkg_metadata)
 
         with pkg_layout.set_dirty_context_manager():
             prev = run_package_method(conanfile, package_id, hook_manager, ref)
