diff --git a/conan/api/subapi/config.py b/conan/api/subapi/config.py
index 22e30fd53..dc0c8c3a0 100644
--- a/conan/api/subapi/config.py
+++ b/conan/api/subapi/config.py
@@ -36,6 +36,16 @@ class ConfigAPI:
     def show(self, pattern):
         return self.global_conf.show(pattern)
 
+    @property
+    def settings_yml(self):
+        """Returns the string contents of the global settings.yml file from the cache folder.
+        If it doesn't exist, returns the default settings.yml string."""
+        from conans.client.conf import default_settings_yml
+        settings_path = HomePaths(self.conan_api.cache_folder).settings_path
+        if os.path.exists(settings_path):
+            return load(settings_path)
+        return default_settings_yml
+
     @property
     def global_conf(self):
         """ this is the new global.conf to replace the old conan.conf that contains
diff --git a/conans/test/integration/conan_api/test_config.py b/conans/test/integration/conan_api/test_config.py
new file mode 100644
index 000000000..76ccd451c
--- /dev/null
+++ b/conans/test/integration/conan_api/test_config.py
@@ -0,0 +1,22 @@
+import os
+from conan.api.conan_api import ConanAPI
+from conans.client.conf import default_settings_yml
+from conans.test.utils.tools import TestClient
+
+def test_config_settings_yml():
+    client = TestClient()
+    api = ConanAPI(cache_folder=client.cache_folder)
+    
+    settings_path = os.path.join(client.cache_folder, "settings.yml")
+    os.remove(settings_path)
+    
+    # 1. Default settings.yml when file doesn't exist
+    assert api.config.settings_yml == default_settings_yml
+    
+    # 2. Custom settings.yml
+    custom_settings = "os: [Windows, Linux, Macos]\n"
+    settings_path = os.path.join(client.cache_folder, "settings.yml")
+    with open(settings_path, "w") as f:
+        f.write(custom_settings)
+        
+    assert api.config.settings_yml == custom_settings
